### ISSUE TITLE
Jump out of brackets if function receives no argument.

### ISSUE THREAD

**Issue by MrConnorKenway** [Saghen/blink.cmp#461]
### Feature Description

First thanks your contribution to this blazingly fast plugin!
In current version, if a function doesn't have arguments, blink.nvim still put the cursor in the middle of brackets instead of moving it to the right of close bracket. Maybe we can do this (`|` is the cursor):
```
vec.beg|
vec.begin()|
```
Is this feature easy to implement? Thanks!

**Comment by Goose97** [comment]
Yeah, I would want this too, but I'm afraid that this could not be done easily. AFAIK, completion item from LSP doesn't contain any field to figure the function arity. For example, here's the response for `lua_ls` when completing `math.max`

```
{
  data = {
    id = 122,
    uri = "file:///Users/goose/.config/nvim/lua/plugins/lsp.lua"
  },
  detail = "function",
  documentation = {
    kind = "markdown",
    value = "```lua\nfunction math.max(x: <Number:number>, ...<Number:number>)\n  -> <Number:number>\n```\n\n---\n\n\nReturns the argument with the maximum value, according to the Lua operator `<`.\n\n[View documents](http://www.lua.org/manual/5.1/manual.html#pdf-math.max)"
  },
  insertText = "max",
  insertTextFormat = 2,
  kind = 3,
  label = "max(x, ...)",
  sortText = "0001"
}
```

The only useful field is `label`. In this case, we can guess the function arity based on that. But this behavior is a implementation detail of each language servers.

But I could be wrong. Let's wait for the author to chime in.

**Comment by Saghen** [comment]
The only way I see to do this reliably would be to make a signature help call after completing, to check if the function takes arguments. But this will have to happen asynchronously, similar to using semantic tokens for auto brackets, and would only work well with LSPs that respond quickly. I'll leave this open in case anyone is interested in working on this

**Comment by walkert** [comment]
Came here looking for the same thing. First @Saghen - this is great work! Second, I've noticed an interesting behaviour. Rust (with rust-analyzer) seems to do this by default when using nvim-cmp without auto pairs or any other plugin. Compared say with Python, which does not do this. When I do the same thing in Rust with blink - the parentheses are completed but the cursor is between them. Could it be that the auto-bracket implementation is discarding some output from the LSP (at least in the case of Rust)?

**Comment by Saghen** [comment]
@walkert That's odd, rust seems to move me to the correct position when no arguments are possible. Are you able to create minimal reproduction of this? https://lazy.folke.io/developers#reprolua

**Comment by walkert** [comment]
@Saghen sure - I've just updated to 0.9.0 and using this config, if I complete on something like `Vec::pop()` my cursor is placed like this: `result.pop(_)` instead of `result.pop()_`:

```
return {
    'saghen/blink.cmp',
    -- optional: provides snippets for the snippet source
    dependencies = {
        'rafamadriz/friendly-snippets',
    },
    -- use a release tag to download pre-built binaries
    version = 'v0.*',
    opts = {}, 
}
```

**Comment by Saghen** [comment]
Can you send the full code you used for the reproduction? I'd like to see how you setup the LSP. It's possible this is a matter of using https://github.com/mrcjkb/rustaceanvim

**Comment by Saghen** [comment]
Closing as I don't intend to work around this issue using signature help. If someone wants to try implementing it, I'd be happy to review a PR
